Control carla vehicles using waywiser teleop and waywise autopilot

r2avula opened the pull request:
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

feature

* **What is the current behavior?** (You can also link to an open issue here)

waywiser does not directly support controlling carla vehicle

* **What is the new behavior (if this is a feature change)?**

vehicles in carla can be controlled using waywiser teleop and waywise autopilot

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

* **Other information**:

r2avula committed:
add followpoint and odom_topic to waywise_autopilot

r2avula committed:
launch carla using carla_ros_bridge

r2avula committed:
create waywiser_carla package with carla map->odom tf publisher node

r2avula committed:
reduce twist_keyboard startup angular velocity

r2avula committed:
add emulated angle sensor in waywiser_carla

r2avula committed:
adapt twist keyboard controls for simulation

r2avula committed:
rename file: waywise_autopilot.cpp -> waywise_car_autopilot.cpp

r2avula committed:
refactor waywise_car and waywise_car_autopilot

r2avula committed:
add waywise truck autopilot

r2avula committed:
add purepursuit_radius param in car and truck autopilot nodes

r2avula committed:
setup trailer mavlink and send its yaw to control tower

r2avula committed:
create osm tile server to visualize carla map at controltower

r2avula committed:
bump WayWise

r2avula committed:
rename file: waywise_rover.cpp -> waywise_car.cpp

r2avula committed:
example launch files for carla

r2avula committed:
waywiser_carla: Update README.md

r2avula committed:
Update README.md

r2avula committed:
refactor carla_osm_tile_server

marvind reviewed (approved):
Really nice work, looks good to me! 😎
I will not manage to test, though.

I especially like waywiser_carla/py_scripts/carla_osm_tile_server.py 😊
waywiser_carla/README.md could use a bit more high-level description IMO, what does this do? And maybe an image/video which I know exists.

r2avula committed:
carla_osm_tile_server: with orthographic projection of aerial images

r2avula committed:
waywiser_carla: add waywiser_twist_to_carla_control node

r2avula committed:
waywiser_teleop: improves twist_keyboard speed update logic

r2avula committed:
Update example launch files for carla

r2avula committed:
waywiser_carla: Update README.md

r2avula commented:
> Really nice work, looks good to me! 😎 I will not manage to test, though.
> 
> I especially like waywiser_carla/py_scripts/carla_osm_tile_server.py 😊 waywiser_carla/README.md could use a bit more high-level description IMO, what does this do? And maybe an image/video which I know exists.

Thanks, its a good suggestion.  I added some description of the carla_osm_tile_server along with a vide. I also updated the carla_osm_tile_server.py to do orthographic projection of camera images to reduce some distortions of buildings.
